Fix codegen for TS for natives that return values by pointers

DeForce opened the pull request:
This fixes native typing TS documentation:
Sample natives that has this issue:

https://runtime.fivem.net/doc/natives/?_0x2975C866E6713290
GetEntityPlayerIsFreeAimingAt
```
# C Declaration:
// GetEntityPlayerIsFreeAimingAt
BOOL GET_ENTITY_PLAYER_IS_FREE_AIMING_AT(Player player, Entity* entity);

# TS Declaration
/**
 * Returns TRUE if it found an entity in your crosshair within range of your weapon. Assigns the handle of the target to the *entity that you pass it.
 * Returns false if no entity found.
 */
declare function GetEntityPlayerIsFreeAimingAt(player: number, entity: number): number;
```

Other examples, just runtime links:
https://runtime.fivem.net/doc/natives/?_0x2E1202248937775C
https://runtime.fivem.net/doc/natives/?_0xDC16122C7A20C933

As you can see natives that has pointers in the declaration, are supposed to be return values, at least for natives that I saw.

The proposed fix fixes natives function/documentation generation so that it makes first pointers returnable.

Fixed Typing:
```
# TS Declaration
declare function GetEntityPlayerIsFreeAimingAt(player: number): [number, number];
```

This is only documentation generation fix, from what I saw - the function actually return array, so it's only documentation issue.

I am not sure about Lua/JS, if they have the same issue or not, but TS would be correctly typed after this PR. At least for the natives that I found have issues.

blattersturm commented:
This breaks codegen for functions such as DeleteEntity:

```ts
declare function DeleteEntity(): number;
```

DeForce commented:
Okay, back to planning board then :)

blattersturm commented:
I suspect this might be hard to correctly implement due to there being no explicit data for `in`/`out`/`inout` arguments like such, which might be why the initial change was a tad weird.

DeForce commented:
I see two options that we can go for:
1. Make return to be unknown[]
```ts
declare function GetAmmoInClip(ped: number, weaponHash: string | number, ammo: number): unknown[];
```

That will make at least typescript not yell at you for wrong typing, because the typing is technically actually unknown.
Also could make the pointer a optional value:
```ts
declare function GetAmmoInClip(ped: number, weaponHash: string | number, ammo?: number): unknown[];
```

2.
Generate two or more functions for each pointer:
```ts
declare function GetAmmoInClip(ped: number, weaponHash: string | number): [number, number];
declare function GetAmmoInClip(ped: number, weaponHash: string | number, ammo:number): number;
```

That would solve the problem by making generation logic a little bit harder. But at least there would be correct typings, but one of the functions would be wrong, which will lead to ambiguity. At least it will have an option to return correctly with second one.

@blattersturm thoughs?

blattersturm commented:
At this time, optional values seem like the better option. It might be confusing for, say, DeleteEntity and break natives where a pointer _isn't_ the last argument, but until someone makes a pass over inout annotations that's probably the safest bet.

DeForce commented:
@blattersturm 
PR has been updated to what we have talked about above.
I have added a comment/warning, that is not really part of the doc-string, I hope that this is acceptable.
Actually, functions that have only one input parameter and doesn't have a set return value will be fine with what I implemented:
```ts
/**
 * Deletes the specified entity, then sets the handle pointed to by the pointer to NULL.
 * @param entity The entity to delete.
 */
declare function DeleteEntity(entity: number): void;


...
 */
declare function GetEntityPopulationType(entity: number): number;


// Return is unknown[] due to pointer value being input of the function
declare function GetAmmoInClip(ped: number, weaponHash: string | number, ammo?: number): unknown[];

/**
 * Returns TRUE if it found an entity in your crosshair within range of your weapon. Assigns the handle of the target to the *entity that you pass it.
 * Returns false if no entity found.
 */
// Return is unknown[] due to pointer value being input of the function
declare function GetEntityPlayerIsFreeAimingAt(player: number, entity?: number): unknown[];
```

gottfriedleibniz commented:
Another edge-case would be where a pointer isn't the only or last argument, e.g., `0xBBE7648349B49BE8`. Just ensure your changes don't clobber native(s) of that form.

DeForce commented:
> Another edge-case would be where a pointer isn't the only or last argument, e.g., `0xBBE7648349B49BE8`. Just ensure your changes don't clobber native(s) of that form.

0xBBE7648349B49BE8 is the same in my result and current.
